suite/tests/linux/signal_pre_syscall.c

```diff
@@ -0,0 +1,408 @@
+/* **********************************************************
```

Comment:
So this test is able to reproduce the case of an interrupted block. What does that look like?
```
instr
syscall_instr
syscall_marker
timestamp
cpu
syscall_instr
syscall_marker
```

or

```
instr
syscall_instr
timestamp
cpu
syscall_instr
syscall_marker
```

test_duplicate_syscalls tests the first case. My suggestion in the other comment thread was to also add a test for the second.

Reply:
The 2nd is tested in the false syscall test (after I removed the instr before the 2nd syscall instr).

For what the signal_pre_syscall test produces: I filed #6128 to document this better and pasted what the test has there.  If we set the app to SIG_IGN it will look like the 2nd one above.